### PR TITLE
Support JSON Access Log format

### PR DESCRIPTION
**Pull request by renuka-fernando**
### Purpose
$subject

#### Sample Logs
```json
{"du":143,"usrAgt":"curl/7.79.1","proto":"HTTP/1.1","fwdFor":null,"reqTxDu":93,"reqPath":"/testkey","host":"enforcer","reqId":"74ffe478-52e2-4a61-bb84-bce35cf0aeee","resDu":143,"resTxDu":0,"bytRec":15,"gwPath":null,"resCd":200,"upsHost":"172.20.0.4:9001","upsSvcTime":null,"t":"2023-07-07T10:12:56.078Z","resFlg":"-","reqDu":2,"rtName":"system#/testkey","gwHost":null,"m":"POST","bytSnt":762,"resCdInf":"via_upstream"}
{"resDu":4294,"fwdFor":null,"reqId":"168a1cf4-af51-44b7-aea8-5b4c035f87b0","upsHost":"54.204.94.184:443","upsSvcTime":null,"m":"GET","reqDu":0,"bytSnt":460,"resFlg":"-","reqPath":"/anything/pet/findByStatus?status=pending","resCdInf":"via_upstream","du":4295,"bytRec":0,"reqTxDu":1418,"rtName":"648957e7a675c35145393369","resCd":200,"t":"2023-07-07T10:13:09.413Z","host":"httpbin.org","resTxDu":0,"proto":"HTTP/1.1","gwPath":"/v2/pet/findByStatus?status=pending","gwHost":"localhost:9095","usrAgt":"curl/7.79.1"}
{"gwPath":"/v2/pet/findByStatus?status=pending","m":"GET","resTxDu":null,"bytSnt":86,"rtName":"648957e7a675c35145393369","fwdFor":null,"resDu":null,"resFlg":"UH","upsSvcTime":null,"reqDu":0,"resCd":503,"upsHost":null,"gwHost":"localhost:9095","host":"localhost:9095","reqTxDu":null,"resCdInf":"no_healthy_upstream","usrAgt":"curl/7.79.1","t":"2023-07-07T10:13:19.223Z","bytRec":0,"reqId":"2f8e9875-2e74-48c7-8560-4feccae92a08","proto":"HTTP/1.1","reqPath":"/v2/pet/findByStatus?status=pending","du":62}
{"rtName":"648957e7a675c35145393369","gwPath":"/v2/pet/findByStatus?status=pending","resCdInf":"via_upstream","bytSnt":482,"du":4257,"reqId":"c566f759-3cea-455a-837b-012565792df6","gwHost":"localhost:9095","t":"2023-07-07T10:13:40.000Z","reqPath":"/anything/pet/findByStatus?status=pending","usrAgt":"My User Agent \"Renuka\" set this","fwdFor":null,"bytRec":0,"resDu":4256,"reqDu":0,"upsSvcTime":null,"reqTxDu":838,"upsHost":"54.210.149.139:443","resFlg":"-","resTxDu":0,"resCd":200,"host":"httpbin.org","proto":"HTTP/1.1","m":"GET"}
```

Config
```toml

[accessLogs.jSONFormat]
t = "%START_TIME%"
gwHost = "%DYNAMIC_METADATA(envoy.filters.http.ext_authz:originalHost)%"
host = "%REQ(:AUTHORITY)%"
m = "%REQ(:METHOD)%"
gwPath = "%DYNAMIC_METADATA(envoy.filters.http.ext_authz:originalPath)%"
reqPath = "%REQ(:PATH)%"
rtName = "%ROUTE_NAME%"
prtcol = "%PROTOCOL%"
resCd = "%RESPONSE_CODE%"
resCdInf = "%RESPONSE_CODE_DETAILS%"
resFlg = "%RESPONSE_FLAGS%"
usrAgt = "%REQ(USER-AGENT)%"
reqId = "%REQ(X-REQUEST-ID)%"
fwdFor = "%REQ(X-FORWARDED-FOR)%"
upsHost = "%UPSTREAM_HOST%"
bytRec = "%BYTES_RECEIVED%"
bytSnt = "%BYTES_SENT%"
du = "%DURATION%"
upsSvcTime = "%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%"
reqTxDu = "%REQUEST_TX_DURATION%"
resTxDu = "%RESPONSE_TX_DURATION%"
reqDu = "%REQUEST_DURATION%"
resDu = "%RESPONSE_DURATION%"
```

### Issues

Fixes #

### Automation tests
 - Unit tests added: N/A
 - Integration tests added: N/A

### Tested environments

Tested - Docker Compose

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/pkg/config/log_types.go [hunk 1]
```diff
@@ -57,15 +59,41 @@ func getDefaultLogConfig() *LogConfig {
 		AccessLogs: &accessLog{
 			Enable:  false,
 			LogFile: "/dev/stdout",
+			LogType: "text",
 			// Following default value of "ReservedLogFormat" is document in log_config.toml for references.
 			// Update log_config.toml if any changes are done here.
-			ReservedLogFormat: "[%START_TIME%]' '%ROUTE_NAME%' '%DYNAMIC_METADATA(envoy.filters.http.ext_authz:originalHost)%' " +
+			ReservedLogFormat: "[%START_TIME%]' '%DYNAMIC_METADATA(envoy.filters.http.ext_authz:originalHost)%' " +
```
Comment (VirajSalaka) [inline adapter/pkg/config/log_types.go:65]: why have we removed the route Name here.

Comment (renuka-fernando) [reply adapter/pkg/config/log_types.go:65]: If we add a new field in the middle of the access log, the fields right to it get shifted. So there would be two positions for old and new logs. If we are adding, we have to append it to the end.

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230710.17

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230710.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230710.3

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20230710.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230710.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230710.4

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230710.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230710.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : prod-deployment-v2 : 20230710.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230710.4

**Pull request merged**